cmd/mirror.go

```diff
@@ -190,13 +193,81 @@ func delComp(repo, id, version string) error {
 	return nil
 }
 
-// the `repo publish` sub command
+// the `mirror modify` sub command
+func newMirrorModifyCmd() *cobra.Command {
+	var privPath string
+	endpoint := environment.Mirror()
+	desc := ""
+	standalone := false
+	hidden := false
+	yanked := false
+
+	cmd := &cobra.Command{
+		Use:  "modify <comp-name> [flags]",
+		Long: "modify component attributes (hidden, standalone, yanked)",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				return cmd.Help()
+			}
+			env := environment.GlobalEnv()
+			if privPath == "" {
+				privPath = env.Profile().Path(localdata.KeyInfoParentDir, "private.json")
+			}
+
+			// Get the private key
+			f, err := os.Open(privPath)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+
+			ki := v1manifest.KeyInfo{}
+			if err := json.NewDecoder(f).Decode(&ki); err != nil {
+				return err
+			}
+
+			m, err := env.V1Repository().FetchComponentManifest(args[0])
+			if err != nil {
+				return err
+			}
+
+			e := remote.NewEditor(endpoint, args[0]).WithDesc(desc)
+			flagSet := set.NewStringSet()
+			cmd.Flags().Visit(func(f *pflag.Flag) {
+				flagSet.Insert(f.Name)
+			})
+			if flagSet.Exist("standalone") {
```

Comment:
Why not check the bool variable?

Reply:
If a bool value (eg. the --hide flag), is false, we don't know if it's `tiup mirror modify xxx --hide=false` or `tiup mirror modify xxx`. These are two very different semantics, the first one is that we should unhide the component, the second one is that we should do nothing on the hidden field